### PR TITLE
fix(medusa): model loader with customizations

### PR DESCRIPTION
**Pull request by adrien2p**

### CODE REVIEW

File: packages/medusa/src/loaders/models.ts [hunk 1]
```diff
@@ -46,6 +47,27 @@ export default (
...
+    if (loaded) {
+      const name = formatRegistrationName(modelPath)
+      const mappedExtensionModel = modelExtensionsMap.get(name)
+      if (mappedExtensionModel) {
+        const coreModel = require(modelPath)
+        const modelName = upperCaseFirst(
+          formatRegistrationNameWithoutNamespace(modelPath)
+        )
+
+        coreModel[modelName] = mappedExtensionModel
```
Comment (olivermrbl) [inline packages/medusa/src/loaders/models.ts]: question: Can we do `loaded[modelName] = mappedExtensionModel` instead and save a file read?

File: packages/medusa/src/loaders/plugins.ts [hunk 2]
```diff
@@ -579,57 +572,76 @@ function registerRepositories(
...
+    if (loaded) {
+      const name = formatRegistrationName(modelPath)
+      const mappedExtensionModel = modelExtensionsMap.get(name)
+      if (mappedExtensionModel) {
+        const coreModel = require(modelPath)
+        const modelName = upperCaseFirst(
+          formatRegistrationNameWithoutNamespace(modelPath)
+        )
+
+        coreModel[modelName] = mappedExtensionModel
```
Comment (olivermrbl) [inline packages/medusa/src/loaders/plugins.ts]: ibid

File: packages/medusa/src/loaders/models.ts (hunk 1 above)
Comment (adrien2p) [reply packages/medusa/src/loaders/models.ts]: done all

File: packages/medusa/src/loaders/plugins.ts [hunk 3]
```diff
@@ -579,57 +572,75 @@ function registerRepositories(
  *    version, id, resolved path, etc. See resolvePlugin
  * @param {object} container - the container where the services will be
  *    registered
+ * @param rootDirectory
+ * @param rootDirectory
```
Comment (riqwan) [inline packages/medusa/src/loaders/plugins.ts]: Copy pasta here

Comment (adrien2p) [reply packages/medusa/src/loaders/plugins.ts]: oupla, nice catch 👍

**Review by riqwan — APPROVED** [review]

**Comment by adrien2p** [comment]
@olivermrbl @riqwan should I merge?

**Comment by riqwan** [comment]
LGTM. You'll need one more approve, best to ask someone else since Oli is on vacation.

**Review by fPolic — APPROVED** [review]
LGTM!

**Pull request merged**